Handle disconnected interfaces more cleanly

j1ah0ng opened the pull request:
This is a continuation of #1515.

j1ah0ng committed:
Handle disconnected interfaces more cleanly

sushisharkjl commented:
Currently implementing logging as well as improved handling and differentiation of (network down but interface present) vs. (interface not present).

j1ah0ng committed:
Merge remote-tracking branch 'upstream/master' into handle-disconnected-interfaces

j1ah0ng committed:
Implementing a flag that enables setting dealing with absent interfaces as being functionally present.

This is useful if ephemeral interfaces exist, ie. if interfaces are not guaranteed to be present
throughout the entire time polybar is being run. Setting the flag to true enables polybar to
continue polling absent interfaces.

It's possible to add a `ignore-not-present` boolean field to the polybar config file under the
specific network interface to enable this feature.

j1ah0ng committed:
Adding the ability to specify different format strings in the case that the interface is disconnected

sushisharkjl commented:
Pending testing, I've added the ability to specify custom `format-absent` and `label-absent` format strings for disconnected interfaces.

j1ah0ng committed:
Adding a case for tunnel pseudo-adapters

chxp82q reviewed (approved):
Looks good :+1:
